runtime/queries/column_timeseries.go

```diff
@@ -121,34 +121,38 @@ func (q *ColumnTimeseries) Resolve(ctx context.Context, rt *runtime.Runtime, ins
 			return err
 		}
 
-		args = append([]any{timezone, timezone, timeRange.Start.AsTime(), timezone, timezone, timeRange.End.AsTime(), timezone, timezone}, args...)
+		args = append([]any{timezone, timeRange.Start.AsTime(), timezone, timeRange.End.AsTime(), timezone}, args...)
+		args = append(args, timezone)
 
 		querySQL := `CREATE TEMPORARY TABLE ` + temporaryTableName + ` AS (
 			-- generate a time series column that has the intended range
 			WITH template as (
-			SELECT
-				range as ` + tsAlias + `
-			FROM
-				range(
-				timezone(?, date_trunc('` + dateTruncSpecifier + `', timezone(?, ?::TIMESTAMPTZ))),
-				timezone(?, date_trunc('` + dateTruncSpecifier + `', timezone(?, ?::TIMESTAMPTZ))),
-				INTERVAL '1 ` + dateTruncSpecifier + `')
+				SELECT
+					range as ` + tsAlias + `
+				FROM
+					range(
+					date_trunc('` + dateTruncSpecifier + `', timezone(?, ?::TIMESTAMPTZ)),
+					date_trunc('` + dateTruncSpecifier + `', timezone(?, ?::TIMESTAMPTZ)),
+					INTERVAL '1 ` + dateTruncSpecifier + `')
```

Comment:
Could it just do `SELECT DISTINCT` here instead of aggregating and using `LAST`?

Reply:
We can do it after final timezone conversion only.